addons/payment_mollie/models/payment_transaction.py

```diff
@@ -39,7 +39,10 @@ def _get_specific_rendering_values(self, processing_values):
         # The acquirer reference is set now to allow fetching the payment status after redirection
         self.acquirer_reference = payment_data.get('id')
 
-        return {'api_url': payment_data["_links"]["checkout"]["href"]}
+        checkout_url = payment_data["_links"]["checkout"]["href"]
```

Comment:
```suggestion
        # Extract the checkout URL from the payment data and add it with its query parameters to the
        # rendering values. Passing the query parameters separately is necessary to prevent them
        # from being stripped off when redirecting the user to the checkout URL, which can happen
        # when only one payment method is enabled on Mollie and query parameters are provided.
        checkout_url = payment_data["_links"]["checkout"]["href"]
```

Reply:
@AntoineVDV updated code according to your feedback :+1: